### DISCUSSION TITLE
Not force to add new line for chained function call after lambda parameter for CascadingCallWrapping rule

### DISCUSSION THREAD

**Discussion by DHosseiny** [detekt/detekt#5591] [General]
## Expected Behavior of the rule
CascadingCallWrapping is a useful rule, thanks for writing this. But there is a case this is annoying. Example:
```
tabsToDeepLinks
    .toList()
    .firstOrNull { pair ->
        pair.second == toUri
    }?.let { pair ->
        switchTab(tabIndex = pair.first)
        return true
    }
```
The `let` function is after a lambda so in this case, I think it is acceptable to not be in a new line. Generally speaking, let the chained call be in the same line with the previous function's lambda end if the previous function's lambda is not one-lined. At least add a Configuration option for this case.

## Context
This is useful to chain function at end of the previous function lambda because work can be done with fewer lines of code without losing readability.

**Comment by schalkms** [comment]
opinion @BraisGabin 

> I don't agree with that but please, open an issue so other people can give their opinions

**Comment by schalkms** [comment]
Thanks for your suggestion! I converted this item, since it's more or less under discussion.

I think we shouldn’t have such opinionated rule behavior. Furthermore, such special behavior creates more maintenance effort in the long run.
Nevertheless, you can still have the suggested behavior in a custom rule by copying the main rule in the detekt repo and extending it.

**Comment by cortinico** [comment]
> The let function is after a lambda so in this case, I think it is acceptable to not be in a new line

To ease the discussion: @DHosseiny could you add expected/actual snippets?

**Comment by atulgpt** [comment]
We should handle this case, since we handled the case https://github.com/detekt/detekt/issues/5598
